test/e2e/framework/autoscaling/autoscaling_utils.go

```diff
@@ -416,6 +443,22 @@ func (rc *ResourceConsumer) GetReplicas() int {
 			framework.Failf(rsIsNil)
 		}
 		return int(rs.Status.ReadyReplicas)
+	case KindCRD:
+		deployment, err := rc.clientSet.AppsV1().Deployments(rc.nsName).Get(context.TODO(), rc.name, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		if deployment == nil {
+			framework.Failf(deploymentIsNil)
+		}
+		deploymentReplicas := int(deployment.Status.ReadyReplicas)
+
+		scale, err := rc.scaleClient.Scales(rc.nsName).Get(context.TODO(), schema.GroupResource{Group: crdGroup, Resource: crdNamePlural}, rc.name, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		crdInstance, err := rc.resourceClient.Get(context.TODO(), rc.name, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		// Update custom resource's status.replicas with child Deployment's current number of ready replicas.
+		framework.ExpectNoError(unstructured.SetNestedField(crdInstance.Object, fmt.Sprint(deploymentReplicas), "status", "replicas"))
```

Comment:
this isn't actually persisting the updated crdInstance... what is the intent behind setting this?

Reply:
True, I've added Update call, thanks for spotting that. The idea behind this was to update the scale subresource, so when e.g. GetReplicasFromRuntimeObject calls CRD instance, we return proper amount of replicas (IIUC correctly this won't update by itself - child Deployment's number of replicas will be updated, but not CRD instance scale subresource).